src/main/java/net/neoforged/neoform/runtime/actions/ExtensibleClasspath.java

```diff
@@ -33,12 +33,25 @@ public List<ClasspathItem> getEffectiveClasspath() {
 
     public void addMinecraftLibraries(Collection<MinecraftLibrary> libraries) {
         for (var library : libraries) {
-            if (library.rulesMatch() && library.getArtifactDownload() != null) {
+            if (library.rulesMatch() && library.getArtifactDownload() != null && !containsLibrary(library)) {
```

Comment:
Do we want to apply this here in general, or should we limit the scope to the `RecompileSourcesAction`?

Reply:
Fine by me.